res/res_pjsip_sdp_rtp.c

```diff
@@ -2246,8 +2246,10 @@ static int apply_negotiated_sdp_stream(struct ast_sip_session *session,
 		ast_rtp_instance_set_timeout(session_media->rtp, session->endpoint->media.rtp.timeout_hold);
 	}
 
-	if (ast_rtp_instance_get_timeout(session_media->rtp)) {
-		session_media->timeout_sched_id = ast_sched_add_variable(sched,	500, rtp_check_timeout,
+	int rtp_timeout = ast_rtp_instance_get_timeout(session_media->rtp);
+
+	if (rtp_timeout) {
+		session_media->timeout_sched_id = ast_sched_add_variable(sched,	rtp_timeout*1000, rtp_check_timeout,
```

Comment:
Looking at the commit history, the code was originally this and was changed in:

https://github.com/asterisk/asterisk/commit/6624e34580f5f5c39d3dbbfb47ba29839397f3d7

Were the changes done in rtp_check_timeout insufficient? How so?

Reply:
https://github.com/asterisk/asterisk/commit/6624e34580f5f5c39d3dbbfb47ba29839397f3d7 does not say the reason to why initial timer value changed to a fixed 500ms, but I believe it introduced a bug. 

1. Invite received. create_rtp is invoked setting rtp_rtp->last_rx = time(NULL)
2. Let it ring for a duration longer than the rtp_timeout setting.
3. Answer the call. sdp is negotiated and scheduler is setup to trigger rtp_check_timeout in 500ms.
4. 500ms elapses without any rtp or rtcp packages received.
5. rtp_check_timeout is triggered by the scheduler. The call is now in state "UP" and the elapsed check is done using the initial setup of last_rx, done when the invite received (in step 1 above). The elapsed time is calculated from the wrong starting point, making the elapsed time greater than rtp_timeout, causing a hang up half a second after it's been answered! 

When the call is answered, sdp negotiated, it should not hangup before rtp_timeout elapsed. Therefor it does not make sense to invoke rtp_check_timeout the first time before the rtp_timeout elapsed.

Example of the problem:
Given rtp_timeout is 10 seconds.
If the call is answered immediately we would have 10 seconds to receive the first rtp before "softhangup"
If the call is answered after 5 seconds we would have 5 seconds to receive the first rtp before "softhangup"
If the call is answered after 15 seconds we would have 0.5 seconds to receive the first rtp before "softhangup"

Conclusion it should not matter how long the call was in the ringing ("UP") state for the rtp timeout logic to trigger the hangup.

My solution would perform the first rtp check after 10 seconds from when the call was answered (=sdp negotiated)